src/util/StrandedPriorityQueue.hpp

```diff
@@ -0,0 +1,98 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2025, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include "util/async/AnyStrand.hpp"
+
+#include <functional>
+#include <optional>
+#include <queue>
+#include <utility>
+#include <vector>
+
+namespace util {
+
+/**
+ * @brief A wrapper for std::priority_queue that serialises operations using a strand
+ * @note This may be a candidate for future improvements if performance proves to be poor (e.g. use a lock free queue)
+ */
+template <typename T, typename Compare = std::less<T>>
+class StrandedPriorityQueue {
+    util::async::AnyStrand strand_;
+    std::priority_queue<T, std::vector<T>, Compare> queue_;
+
+public:
+    /**
+     * @brief Construct a new priority queue on a strand
+     * @param strand The strand to use
+     */
+    StrandedPriorityQueue(util::async::AnyStrand&& strand) : strand_(std::move(strand))
```

Comment:
Shouldn't there be a size limit for the queue to prevent overload?

Reply:
Yeah, probably. I can add something 👍